### DISCUSSION TITLE
bms in cobol project

### DISCUSSION THREAD

**Discussion by guocht** [IBM/zopeneditor-about#161] [Q&A]
Hi,
We have BMS files(separated directory) in the cobol project. How I can config the open editor to upload the bms file to uss when use "user build" function and generate the copy book automatically when compile the cobol file?

**Comment by FALLAI-Denis** [comment]
Hi,
BMS sources are basically made up of Assembler macros.
Do you use the ZOE editor dedicated to the HLASM language to manipulate your BMS sources in VS Code?
Have you studied the possibility of implementing a "wysiwyg" editor to handle your BMS sources in VS Code (note: also valid for reports editing)?

**Reply by guocht** [reply]
I did not want to edit the BMS the file. I just want the file can be upload and compile correctly when compile the cobol files.

**Reply by FALLAI-Denis** [reply]
Yes, I understood your request correctly.
My question was on the sidelines of your request and was just to know and get an opinion on how you were working on your BMS source because we have an ongoing project on this.

**Reply by guocht** [reply]
For BMS edit, team has no specific tool to handle this, As  no editor give content level assistant.

**Comment by phaumer** [comment]
@guocht the workspace settings have an `additionalDependencies` property that allows you to list additional folders and files that should be uploaded to the workspace. The samples use it for the dbb-zappbuild properties files, but you can upload other sources with it.

https://www.ibm.com/docs/en/wdfrhcw/1.3.0?topic=code-setting-up-user-build
